### PR TITLE
Fix: 561 - ctTSO NTC should be run

### PR DESCRIPTION
**Pull request by reisingerf**
Add `must_be_tumor_or_ntc` method to allow runs on negative controls. Add new test case accordingly.

### CODE REVIEW

**Comment by alexiswl** [comment]
Ah I was just about to push the same thing!

File: data_processors/pipeline/domain/workflow.py [hunk 1]
```diff
@@ -353,6 +353,14 @@ def must_be_tumor(self):
             raise LabMetadataRuleError(f"Not tumor sample.")
         return self
 
+    def must_be_tumor_or_ntc(self):
+        """Sample Phenotype must be tumor"""
+        from data_portal.models.labmetadata import LabMetadataPhenotype
+        if self.this_metadata.phenotype.lower() == LabMetadataPhenotype.TUMOR.value.lower() or self.this_metadata.phenotype.lower() == LabMetadataPhenotype.N_CONTROL.value.lower():
```
Comment (alexiswl) [inline data_processors/pipeline/domain/workflow.py:359]: I think for consistency this should be `if ... then fail else return self`

The following would negate this logic:

```python
if not (
    self.this_metadata.phenotype.lower() == LabMetadataPhenotype.TUMOR.value.lower() or 
    self.this_metadata.phenotype.lower() == LabMetadataPhenotype.N_CONTROL.value.lower()
    ):
    raise LabMetadataRuleError(f"Not tumor sample or NTC sample.")
return self

```

File: data_processors/pipeline/domain/workflow.py [hunk 2]
```diff
@@ -353,6 +353,14 @@ def must_be_tumor(self):
...
+            return self
+        else:
+            raise LabMetadataRuleError(f"Not tumor sample.")
```
Comment (alexiswl) [inline data_processors/pipeline/domain/workflow.py:362]: "Not tumor or NTC sample"

File: data_processors/pipeline/orchestration/tests/test_dragen_tso_ctdna_step.py [hunk 3]
```diff
@@ -132,6 +132,63 @@ def test_dragen_tso_ctdna(self):
...
+        when(libwes.WorkflowRunsApi).get_workflow_run(...).thenReturn(mock_wfl_run)
+
+        # LPRJ200438
+        mock_labmetadata_tumor = LabMetadata()
+        mock_labmetadata_tumor.subject_id = tn_mock_subject_id
+        mock_labmetadata_tumor.library_id = mock_library_id
+        mock_labmetadata_tumor.phenotype = LabMetadataPhenotype.N_CONTROL.value
+        mock_labmetadata_tumor.type = LabMetadataType.CT_DNA.value
+        mock_labmetadata_tumor.assay = LabMetadataAssay.CT_TSO.value
+        mock_labmetadata_tumor.workflow = LabMetadataWorkflow.RESEARCH.value
```
Comment (alexiswl) [inline data_processors/pipeline/orchestration/tests/test_dragen_tso_ctdna_step.py:168]: Workflow for a NTC sample will be CONTROL

**Comment by victorskl** [comment]
Thkz both; will review this soon.

**Review by alexiswl — APPROVED** [review]

**Review by victorskl — APPROVED** [review]
Yup. This is good! Thanks again.

**Pull request merged**